include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -101,24 +101,35 @@ struct CollectiveMma<
   static constexpr int SubgroupSize = DispatchPolicy::SubgroupSize;
 
   using MmaAtomShape = typename TiledMma::AtomShape_MNK;
-  using SubgroupTileShape = decltype(tile_shape(TiledMma()));
 
-  static constexpr auto sg_per_wg_m = get<0>(WorkgroupTileShape{}) / get<0>(SubgroupTileShape{});
-  static constexpr auto sg_per_wg_n = get<1>(WorkgroupTileShape{}) / get<1>(SubgroupTileShape{});
-
-  static constexpr uint32_t MaxThreadsPerBlock =
-          cute::size(WorkgroupTileShape{}) / cute::size(SubgroupTileShape{}) * SubgroupSize;
-
-  static constexpr int FragsM = get<0>(SubgroupTileShape{}) / get<0>(MmaAtomShape()); // A frags per sub_group
-  static constexpr int FragsN = get<1>(SubgroupTileShape{}) / get<1>(MmaAtomShape()); // B frags per sub_group
-  static constexpr int FragsK = get<2>(SubgroupTileShape{}) / get<2>(MmaAtomShape());
-
-  // Calculate the vector width based on the amount of registers 
-  // required per work item by dividing the total fragment size by 
-  // the sub_group size.
-  static constexpr int VecC = (get<1>(MmaAtomShape()) * get<0>(MmaAtomShape())) / SubgroupSize;
-  static constexpr int VecA = (get<0>(MmaAtomShape()) * get<2>(MmaAtomShape())) / SubgroupSize;
-  static constexpr int VecB = (get<1>(MmaAtomShape()) * get<2>(MmaAtomShape())) / SubgroupSize;
+  static constexpr auto BLK_M = get<0>(WorkgroupTileShape{});
+  static constexpr auto BLK_N = get<1>(WorkgroupTileShape{});
+  static constexpr auto BLK_K = get<2>(WorkgroupTileShape{});
+  
+  static constexpr auto ATOM_M = get<1>(typename TiledMma::ThrLayoutVMNK{}.shape());
+  static constexpr auto ATOM_N = get<2>(typename TiledMma::ThrLayoutVMNK{}.shape());
+  static constexpr auto ATOM_K = get<3>(typename TiledMma::ThrLayoutVMNK{}.shape());
+
+  static constexpr auto SG_M = ceil_div(BLK_M, ATOM_M);
+  static constexpr auto SG_N = ceil_div(BLK_N, ATOM_N);
+  static constexpr auto SG_K = ceil_div(BLK_K, ATOM_K);
+  using SubgroupTileShape = Shape<decltype(SG_M), decltype(SG_N), decltype(SG_K)>;
```

Comment:
This does not match the tileMMA definition in the gemm example

```c++
using TiledMma = TiledMMA<MMA_Atom<XE_8x16x16_F32BF16BF16F32_TT>,
          Layout<Shape<_8,_2,_1>>,
          Tile<Underscore,Underscore,Underscore>>; // Subgroup level-tile
```

There is some documentation in https://github.com/NVIDIA/cutlass/blob/main/media/docs/cute/0t_mma_atom.md#tiledmmas

Basically, the tileMMA definition expands the MMA operation along the thread dimension, while the implementation here uses that information to expand along the data dimension.

Reply:
I think my declaration matches the definition, actually 8x2 subgroups per workgroup is called, i.e 8x2 xmx

Reply:
And I want to get subgroup size from tileMMA if `Tile<Underscore,Underscore,Underscore>` is default value.

Reply:
In that case the TiledMMA definition should be 

```c++
using TiledMma = TiledMMA<MMA_Atom<XE_8x16x16_F32BF16BF16F32_TT>,
          Layout<Shape<_8,_2,_1>>,
          Tile<_64,_32,_16>>;
```

The implementation in this file should allow users to change the MMA tile size and GEMM should still work. For example, users should be able to use: 

```c++
using TiledMma = TiledMMA<MMA_Atom<XE_8x16x16_F32BF16BF16F32_TT>,
          Layout<Shape<_8,_2,_1>>,
          Tile<_64,_64,_16>>;
```

Performance might drop, but the output should still be correct.

Reply:
Yes, I think it is my understanding. My concern is how to get subgroup size from the declaration of TiledMMA. My solution is first getting the layout subgroups(the second parameter) involved, then computing the subgroup size

Reply:
You can get all the information from the tileMMA object. For example for:

```c++
using TiledMma2 = TiledMMA<MMA_Atom<XE_8x16x16_F32BF16BF16F32_TT>,
        Layout<Shape<_8,_2,_1>>,
        Tile<_64,_64,_32>>; // Subgroup level-tile

  print("AtomShape_MNK: ");
  print(TiledMma2::AtomShape_MNK{});
  print("\n");
  print("ThrLayoutVMNK: ");
  print(TiledMma2::ThrLayoutVMNK{});
  print("\n");
  print("tile_shape: ");
  print(tile_shape(TiledMma2()));
  print("\n"); 
  ```
  
The output would be:
  
```bash
AtomShape_MNK: ( _8,_16,_16) 
# 8x16x16 from XE_8x16x16_F32BF16BF16F32_TT 

ThrLayoutVMNK: ( _16,_8,_2,_1):( _1,_16,_128,_0) 
# the first 16 is the number of threads in XE_8x16x16_F32BF16BF16F32_TT. The rest of from Layout<Shape<_8,_2,_1>>

tile_shape: ( _64,_64,_32) 
# from Tile<_64,_64,_32>
```

Reply:
TileMMA can explicitly supply the info of tile size built by several subgroups，while want I want is subgrouop size per subgroup. I must calculate it simply unless I inject two lines code into TiledCopy